File: Python/Factorial/factorial.py
#! /usr/bin/env python3
# This program finds the factorial of a user specified number

# Define variables
number = 0;
factorial = 0;
counter = 0;

number = int( input( "Enter a nonnegative integer: " ) ) # Read number from user

factorial = number # Assign the value of number to factorial

counter = number - 1 # Assign number minus 1 to counter

# If number is greater than 0
if number > 0:
    # Use a loop to calculate the factorial
    while counter >= 1:
        factorial *= number - counter
        counter -= 1; # Increment counter
elif number == 0: # If number if 0
    # Factorial is 1
    factorial = 1
else: # Otherwise
    print( "Invalid input." ) # Print an error message

print( "The factorial of {} is {}".format( number, factorial ) ) # Print the factorial

File: C++/Factorial/factorial.cpp
// This program finds the factorial of a number
#include <iostream>

// Function main begins program execution
int main()
{
    // Declare variables
    int number;
    int counter;
    int factorial;

    std::cout << "Enter a number: "; // Prompt user for input
    std::cin >> number; // Read number from user

    factorial = number; // Initialize factorial to number
    counter = number - 1; // Initialize counter to number minus 1

    // If number is greater than 0
    if ( number > 0 )
    {
        // While counter is greater than or equal to 1
        while ( counter >= 1 )
        {
            factorial *= number - counter; // Multiply factorial by number minus counter
            counter--; // Decrement counter
        } // end while
    } // end if
    // If number is 0
    else if ( number == 0 )
        factorial = 1; // set factorial to 1

    // If number is greater than or equal to 0
    if ( number >= 0 )
        std::cout << number << " factorial is " << factorial << std::endl; // Print a message
    else // Otherwise
        std::cerr << "Invalid input!" << std::endl; // Print a message

    return 0; // Indicate the program ended successfully
} // end function
